adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java

```diff
@@ -0,0 +1,196 @@
+     *
+     * @param activity     the activity the activity
+     * @param screenNameId from strings resources
+     */
+    public static void sendScreen(Activity activity, @StringRes int screenNameId) {
+        if (screenNameId == 0) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a screenName to send the Screen");
+            return;
+        }
+        sendScreen(activity, activity.getString(screenNameId));
+    }
+
+    /**
+     * Send a screen with screenId
+     *
+     * @param activity     the activity the activity
+     * @param screenNameId from strings resources
+     * @param label        label to format arguments in string resource
+     */
+    public static void sendScreen(Activity activity, @StringRes int screenNameId, String label) {
+        if (screenNameId == 0) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a screenName to send the Screen");
+            return;
+        }
+        if (label == null) {
+            sendScreen(activity, activity.getString(screenNameId));
+        } else {
+            sendScreen(activity, activity.getString(screenNameId, label));
+        }
+    }
+
+    /**
+     * Send a screen with screenName
+     *
+     * @param activity   the activity
+     * @param screenName the screen name
+     */
+    public static void sendScreen(Activity activity, String screenName) {
+
+        if (TextUtils.isEmpty(screenName)) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a screenName to send the Screen");
+            return;
+        }
+
+        if (BuildConfig.DEBUG) {
+            Log.i(FirebaseAnalyticsManager.class.getCanonicalName(), "Setting Screen name: " + screenName);
+        }
+
+        FirebaseAnalytics.getInstance(activity).setCurrentScreen(activity, screenName, null);
+    }
+
+    /**
+     * Send an event with eventName with all event types on bundle
+     *
+     * @param activity    the activity
+     * @param eventNameId name that will appear on timeline
+     * @param eventInfo   info about the event
+     */
+    public static void sendEvent(Activity activity, @StringRes int eventNameId, Map<String, String> eventInfo) {
+        if (eventNameId == 0) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a eventName to send the Event");
+            return;
+        }
+        sendEvent(activity, activity.getString(eventNameId), eventInfo);
+    }
+
+    /**
+     * Send an event with eventName with all event types on bundle
+     *
+     * @param activity  the activity
+     * @param eventName name that will appear on timeline
+     * @param eventInfo info about the event
+     */
+    public static void sendEvent(Activity activity, String eventName, Map<String, String> eventInfo) {
+        if (TextUtils.isEmpty(eventName)) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a eventName to send the Event");
```

Comment:
This `if` is duplicated in other methods. Should be placed in a separated method.